src/ascend.ts

```diff
@@ -40,9 +51,88 @@ export class AscendError extends Error {
       super(`Invalid astral item: ${cause}!`);
     } else if (cause instanceof Class) {
       super(`Invalid class ${cause} for this path!`);
-    } else {
+    } else if (cause instanceof Path) {
       super(`Invalid path ${cause}!`);
-    }
+    } else super(cause);
+    this.cause = cause;
+  }
+}
+
+const worksheds = [
+  "warbear LP-ROM burner",
+  "warbear jackhammer drill press",
+  "warbear induction oven",
+  "warbear high-efficiency still",
+  "warbear chemistry lab",
+  "warbear auto-anvil",
+  "spinning wheel",
+  "snow machine",
+  "Little Geneticist DNA-Splicing Lab",
+  "portable Mayo Clinic",
+  "Asdon Martin keyfob",
+  "diabolic pizza cube",
+  "cold medicine cabinet",
+] as const;
+type Workshed = typeof worksheds[number];
+
+const gardens = [
+  "packet of pumpkin seeds",
+  "Peppermint Pip Packet",
+  "packet of dragon's teeth",
+  "packet of beer seeds",
+  "packet of winter seeds",
+  "packet of thanksgarden seeds",
+  "packet of tall grass seeds",
+  "packet of mushroom spores",
+] as const;
+type Garden = typeof gardens[number];
+
+const eudorae = [
+  "My Own Pen Pal kit",
+  "GameInformPowerDailyPro subscription card",
+  "Xi Receiver Unit",
+  "New-You Club Membership Form",
+  "Our Daily Candles™ order form",
+] as const;
+type Eudora = typeof eudorae[number];
+
+const isWorkshed = createStringUnionTypeGuardFunction(worksheds);
+const isGarden = createStringUnionTypeGuardFunction(gardens);
+const isEudora = createStringUnionTypeGuardFunction(eudorae);
+const isDesk = createStringUnionTypeGuardFunction(desks);
+const isNightstand = createStringUnionTypeGuardFunction(nightstands);
+const isCeiling = createStringUnionTypeGuardFunction(ceilings);
+
+export class AscensionPrepError extends Error {
+  cause: string;
+  constructor(cause: string) {
```

Comment:
I'd change this to take the current value as well as the cause. Then then the string building logic is easier.